File: problems/Number of Substrings With Only 1s/numSub.java
class Solution {
    public int numSub(String s) {
        int result = 0;
        int count = 0;
        int mod = (int)1e9 + 7;

        for (int i = 0; i < s.length(); ++i) {
            count = '1' == s.charAt(i) ? 1 + count : 0;
            result = (result + count) % mod;
        }

        return result;
    }
}

File: problems/Number of Substrings With Only 1s/numSub.cs
public class Solution {
    public int NumSub(string s) {
        var mod = 1000000007;
        var result = 0;
        var count = 0;
        
        foreach (char letter in s) {
            count = '1' == letter ? 1 + count : 0;
            result = (result + count) % mod;
        }
        
        return result;
    }
}
